object_intern.go

```diff
@@ -526,3 +518,66 @@ func (oi *ObjectIntern) Len(ptrs []uintptr) (retLn []int, all bool) {
 	}
 	return
 }
+
+func (oi *ObjectIntern) JoinStrings(nodes []uintptr) (string, error) {
+	lengths, complete := oi.Len(nodes)
+	if !complete {
+		return "", fmt.Errorf("Could not find object in store")
+	}
+
+	var tmpString string
+	var bld strings.Builder
+
+	stringHeader := (*reflect.StringHeader)(unsafe.Pointer(&tmpString))
+
+	first, err := oi.GetStringFromPtr(nodes[0])
```

Comment:
why is the first element a special case which gets its own variable and for which `GetStringFromPtr()` is called?

Couldn't that also be done in that following `for` loop if you'd only prepend the dot to the `node` string if `idx > 0`